pkg/skaffold/util/util.go

```diff
@@ -242,3 +252,15 @@ func NonEmptyLines(input []byte) []string {
 	}
 	return result
 }
+
+// CloneThroughJSON marshals the old interface into the new one
```

Comment:
doesn't necessarily need to happen in this PR, but I think this is the exact same logic we use when calling `config.Convert()` for auto-upgrading configs, if so we should reuse there.

Reply:
for sure, I can switch those to using this function in another PR